compiler/rustc_metadata/src/dependency_format.rs

```diff
@@ -224,7 +224,12 @@ fn calculate_type(tcx: TyCtxt<'_>, ty: CrateType) -> DependencyList {
                     Linkage::Static => "rlib",
                     _ => "dylib",
                 };
-                sess.emit_err(LibRequired { crate_name: tcx.crate_name(cnum), kind: kind });
+                let crate_name = tcx.crate_name(cnum);
+                if crate_name.as_str().starts_with("rustc_") {
```

Comment:
Can we make this a more general check, `if crate_can_be_loaded_from_sysroot(crate_name)` or something like that? I understand that's pretty hard, fine to open a follow-up issue if you don't want to spend time on it.

Reply:
I misunderstood, this error is specifically about having an .rmeta but not an .rlib, which isn't about *sysroot* crates, it's about specifically crates that are in `rustc_driver`. No need to change the error.